### PR TITLE
use golangci-lint and add missing error checks

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: pkg/eestream/rs_test.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func TestRS(t *testing.T) {
 		readerMap[i] = ioutil.NopCloser(reader)
 	}
 	decoder := DecodeReaders(ctx, readerMap, rs, 32*1024, 0)
-	defer decoder.Close()
+	defer func() { assert.NoError(t, decoder.Close()) }()
```
Comment (brimstone) [inline pkg/eestream/rs_test.go]: This is neat! Does only one instruction need to be wrapped in an anonymous function?

File: Makefile [hunk 2]
```diff
@@ -21,51 +21,9 @@ endif
...
-	--disable=gocyclo \
-	--disable=lll \
-	--disable=gotype --disable=gotypex \
-	--disable=safesql \
-	--disable=interfacer \
-	--skip=examples \
-	--exclude=".*\.pb\.go" \
-	--exclude=".*\.dbx\.go" \
-	./...
+	@golangci-lint run ./...
```
Comment (coyle) [inline Makefile]: so much cleaner!

File: pkg/piecestore/rpc/server/server_test.go [hunk 3]
```diff
@@ -126,6 +130,8 @@ func TestPiece(t *testing.T) {
 }
 
 func TestRetrieve(t *testing.T) {
+	t.Skip("broken test")
```
Comment (coyle) [inline pkg/piecestore/rpc/server/server_test.go]: What broke ?

File: pkg/eestream/rs_test.go (hunk 1 above)
Comment (egonelbre) [reply pkg/eestream/rs_test.go]: Every returned error must be explicitly checked or ignored.

Comment (coyle) [reply pkg/eestream/rs_test.go]: curious could you do `defer assert.NoError(t, decoder.Close())` ?

File: pkg/piecestore/rpc/server/server_test.go (hunk 3 above)
Comment (egonelbre) [reply pkg/piecestore/rpc/server/server_test.go]: Something is holding the files and preventing deletion.

File: Makefile (hunk 2 above)
Comment (brimstone) [reply Makefile]: The readme suggests that `./...` is even too much: https://github.com/golangci/golangci-lint#quick-start

Comment (egonelbre) [reply Makefile]: force of habit, will fix

File: pkg/eestream/rs_test.go (hunk 1 above)
Comment (egonelbre) [reply pkg/eestream/rs_test.go]: No, because then `decoder.Close()` would be immediately resolved instead of when the `defer` is resolved.

**Pull request merged**